aws/resource_aws_backup_plan_test.go

```diff
@@ -31,11 +32,13 @@ func TestAccAwsBackupPlan_basic(t *testing.T) {
 					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "backup", regexp.MustCompile(`backup-plan:.+`)),
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
-					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "rule_name", rName),
-					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "target_vault_name", rName),
-					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "schedule", "cron(0 12 * * ? *)"),
-					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "lifecycle.#", "0"),
-					testAccCheckAwsBackupPlanRuleAttr(resourceName, &ruleNameMap, rName, "recovery_point_tags.%", "0"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"rule_name":             rName,
+						"target_vault_name":     rName,
+						"schedule":              "cron(0 12 * * ? *)",
+						"lifecycle.#":           "0",
+						"recovery_point_tags.%": "0",
```

Comment:
originally this attribute was causing the test to fail similar to what happens when `"lifecycle.0.delete_after":       "0",` is checked below as these attributes are not in state for some reason 🤔 the update in `tfawsresource` should account for these

Reply:
You may want to check with the SDK team, but I have a feeling that we will not want to support the legacy flatmap `.#` and `.%` syntax checks for missing/empty attributes in anything new until they come up with a "better" way to do rich type checking in the acceptance testing functions. I'd expect at some point we could be able to write something like:

```
"recovery_point_tags": nil,
```

Or

```
"recovery_point_tags": something.EmptyOrMissingMap(),
```

And the framework will do the right thing. 👍  For here and now I think we can drop these "missing attribute" specific checks.

Reply:
sounds good! now that you mention it, that legacy support for flatmaps/missing attributes does sound familiar especially going back to when these new helper methods were implemented 👍